ff_locking.c

```diff
@@ -163,18 +167,19 @@ void FF_LockDirectory( FF_IOManager_t * pxIOManager )
         return;
     }
 
-    for( ; ; )
+    for (;;) 
     {
-        /* Called when a task want to make changes to a directory.
-         * It waits for the desired bit to come high, and clears the
-         * bit so that other tasks can not take it. */
-        xBits = xEventGroupWaitBits( pxIOManager->xEventGroup,
-                                     FF_DIR_LOCK_EVENT_BITS, /* uxBitsToWaitFor */
-                                     pdTRUE,                 /* xClearOnExit */
-                                     pdFALSE,                /* xWaitForAllBits n.a. */
-                                     FF_TIME_TO_WAIT_FOR_EVENT_TICKS );
-
-        if( ( xBits & FF_DIR_LOCK_EVENT_BITS ) != 0 )
+        xEventGroupWaitBits(pxIOManager->xEventGroup,
```

Comment:
This extra statement was indeed necessary.

Reply:
The return value of xEventGroupWaitBits is not checked here. Is that on purpose?

if the call times out (FF_TIME_TO_WAIT_FOR_EVENT_TICKS ticks elapsed), then the event bit will be erroneously cleared right?